<launch>
  <arg name="node_start_delay" default="10.0" />

  <include file="$(find stage_ros)/launch/stage_pr2_laas_adream.launch">
    <!-- <arg name="fast_mode" value="$(arg fast_mode)"/>
    <arg name="gui" value="$(arg gui)" unless="$(arg fast_mode)"/> -->
  </include>

  <node pkg="tf" type="static_transform_publisher" name="odom_transform_r" args="0 0 0 0 0 0 /map /odom 20"/>
  <node pkg="tf" type="static_transform_publisher" name="odom_transform_h1" args="0 0 0 0 0 0 /map /human1/odom 20"/>
  <node pkg="tf" type="static_transform_publisher" name="odom_transform_h2" args="0 0 0 0 0 0 /map /human2/odom 20"/>
  <!-- <node pkg="fake_localization" type="fake_localization" name="fake_localization" respawn="true" output="screen"/> -->

  <node name="map_server" pkg="map_server" type="map_server" args="$(find cohan_navigation)/maps/laas_adream_stage.yaml"/>

  <!-- Stage agents to /tracked_agents -->
  <node name="agents" pkg="stage_ros" type="agents_bridge.py" args="2" output="screen"/>

  <!-- agent pose prediction, for the local-planning -->
  <node pkg="agent_path_prediction" type="agent_path_prediction" name="agent_path_predict" output="screen" >
    <remap from="/agent_path_predict/tracked_agents" to="/tracked_agents"/>
    <remap from="/agent_path_predict/external_agent_trajs" to="/local_planner/HATebLocalPlannerROS/agent_local_trajs"/>

    <param name="velobs_mul" value="1.0"/>
    <param name="velscale_mul" value="2.0"/>
  </node>

  <node pkg="agent_path_prediction" type="predict_goal.py" name="agent_goal_predict" output="screen"/>

  <!-- Filter the agents from laser scan -->
  <include file="$(find cohan_layers)/launch/agent_filter.launch">
    <arg name="simulator" value="stage"/>
  </include>


  <node pkg="global_planner" type="planner" name="global_planner" output="screen">
    <rosparam file="$(find cohan_navigation)/config/stage_pr2_move_base_config/global_costmap_params.yaml" command="load" ns="costmap"/>
  </node>

  <node pkg="hateb_local_planner" type="local_planner" name="local_planner" output="screen" launch-prefix="bash -c 'sleep 2.0; $0 $@' ">
    <remap from="/move_base/HATebLocalPlannerROS/agents_states" to="/local_planner/hateb_local_planner_test/agents_states" />

    <rosparam file="$(find cohan_navigation)/config/stage_pr2_move_base_config/hateb_local_planner_params.yaml" command="load" ns="hateb_local_planner_test" />
    <rosparam file="$(find cohan_navigation)/config/stage_pr2_move_base_config/local_costmap_params.yaml" command="load" ns="local_costmap"/>
  </node>

  <node pkg="hateb_local_planner" type="get_optim_vel" name="get_optim_vel" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>

</launch>
